code/modules/power/singularity/emitter.dm

```diff
@@ -24,6 +24,7 @@
 	var/frequency = 0
 	var/id_tag = null
 	var/datum/radio_frequency/radio_connection
+	var/datum/effect_system/spark_spread/sparks

```

Comment:
Do you use the dedicated holder here because it's frequent as opposed to the cases you removed?

Reply:
Yup. No sense in making a datum and deleting it constantly, only for it to be created and used, again, in the same manner.